evm/db/state.py

```diff
@@ -49,6 +51,7 @@ def __init__(self, db, root_hash=BLANK_ROOT_HASH, read_only=False):
             self.db = ImmutableDB(db)
         else:
             self.db = db
+        self.db = TrackedDB(self.db)
```

Comment:
Instead of overwriting `self.db` here maybe move this up into the previous `if/else`